clang/test/SemaSYCL/device_global.cpp

```diff
@@ -58,6 +117,7 @@ int main() {
 
     // expect no error on non_const_static declaration if decorated with
     // [[__sycl_detail__::global_variable_allowed]]
+    // expected-error@+1{{'device_global' variables must be static or declared at namespace scope}}
```

Comment:
Is this a valid error? The comment above seems to imply this should not generate an error

Reply:
Indeed. I fixed it now. Thank you both.